flink-metrics/flink-metrics-core/src/main/java/org/apache/flink/metrics/MetricGroup.java

```diff
@@ -186,6 +186,15 @@
      */
     Map<String, String> getAllVariables();
 
+    /**
+     * Returns a map of all variables and their associated value, for example {@code
+     * {"<host>"="host-7", "<tm_id>"="taskmanager-2"}}.
+     *
+     * @param filter character filter which is applied to the variables and values if not null.
+     * @return map of all variables and their associated value
+     */
+    Map<String, String> getAllVariables(CharacterFilter filter);
```

Comment:
Yes, I hesitated on that one. Ok I'll revert the interface extension and related changes.

Reply:
I think we will need it in the `AbstractMetricGroup`.
Ideally no class works directly against the `FrontMetricGroup`; this currently only happens for some reporters because we do not find the time to properly move `getLogicalScope()` into the `MetricGroup` interface.

Reply:
ok clear, thanks